samples/drivers/i2c/rtio_loopback/README.rst

```diff
@@ -0,0 +1,50 @@
+.. zephyr:code-sample:: i2c-rtio-loopback
+   :name: I2C RTIO loopback
+   :relevant-api: rtio
```

Comment:
just noticed but in case this PR gets another update, it feels like adding `i2c_interface` to the list would be nice -- after all, it lives in `samples/drivers/i2c` and it would make sense for this to also show up here https://docs.zephyrproject.org/latest/hardware/peripherals/i2c.html#api-reference

Reply:
added